app/models/business_objective.rb

```diff
@@ -10,4 +10,8 @@ class BusinessObjective < ApplicationRecord
   scope :in_account, lambda { |account_or_account_id|
     joins(:accounts).where(accounts: { id: account_or_account_id })
   }
+
+  def self.from_title(title)
+    find_or_create_by(title: title) if title
```

Comment:
`find_or_create_by` has to be by `title` AND `account_id`, otherwise it will match the BO to any BO in any account

Reply:
In fact the validation above should ideally take that into account...

Reply:
The BusinessObjective model has no account_id:

```rb
BusinessObjective(id: uuid, title: string, created_at: datetime, updated_at: datetime)
```

The BO <-> Account association happens on the profile with BO <-> Profile <-> Account

Reply:
Oh, good point - it should take into account the account in here, perhaps by searching on the `Pundit.policy_scope` of BOs for the current account? Just ` find_or_create_by` is searching *across all accounts*